### ISSUE TITLE
Failed to launch the browser process on Heroku

### ISSUE THREAD

**Issue by averybiskup** [transitive-bullshit/chatgpt-api#119]
I'm trying to host my app using chatgpt-api on Heroku, but puppeteer needs to run headless in order for that to work.

Would someone be able to add an option to run puppeteer headless?

I believe all that needs to change is /src/open-ai.ts line 132. `headless: true,`

**Comment by transitive-bullshit** [comment]
@averybiskup this isn't recommended at the moment until we have a robust solution to bypass captchas. Right now, oftentimes the headful version is very much necessary in order to manually bypass captchas.

**Comment by elreco** [comment]
@transitive-bullshit Do you know if it's possible now to run puppeteer headless? Are you using chatgpt-api from your personal computer so far?

**Comment by transitive-bullshit** [comment]
@elreco I'm currently running minimized puppeteer windows, so not completely headless.

I know some people are using `xvfb` successfully on linux to run with a fake virtual framebuffer.

I have not tried the "headless: true" option to puppeteer yet.

**Comment by LaysDragon** [comment]
yes, I use with google auth provider and lunch a bot with xvfb inside docker. I'm curious,since google auth won't encounter CAPTCHAs,why not let google auth login can run with headless mode?

**Comment by kraj011** [comment]
Hey! Any plans on adding this as an option now that theres 2captcha/nopecha implementation? Ty
